### PR TITLE
Add fighter related info to ship design articles

### PR DESCRIPTION
**Pull request by dbenage-cx**
Suggested by UrshMost in [forum](http://www.freeorion.org/forum/viewtopic.php?f=28&t=10339#p86772).

Adds fighter damage, capacity and launch capacity to pedia article of ship designs.
In lieu of specific info for flak cannons, adds a total number of weapon shots fired (per round).

### CODE REVIEW

**Comment by MatGB** [comment]
I like this, thanks, it makes designing and comparing easier.

How much extra work would it be to add similar auto generated text for the basic hull info entries? Ideally there should be no game info in them that isn't auto generated, so it should list

```
Structure:     Shield: 
Speed:     Fuel Capacity: 
Detection Range:     Stealth: 
```
as well as ideally number of each slot type, that would let us then remove a lot of the badly worded text I put in as a placeholder years ago.

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -4146,19 +4111,46 @@ Parts: %3%
...
 '''
-For species: %18%
-Total [[encyclopedia DAMAGE_TITLE]]: %17%    [[metertype METER_STRUCTURE]]: %8%    [[metertype METER_SHIELD]]: %9%
-[[metertype METER_SPEED]]: %13%    [[metertype METER_FUEL]] Capacity: %14%
-[[metertype METER_DETECTION]]: %10%    [[metertype METER_STEALTH]]: %11%
-Colonization Capacity: %15%    [[metertype METER_TROOPS]]: %16%
-Estimated combat strength (ignoring shields):  %19$.f   (%20$.2f per PP)
-Against enemy attack %21$.1f and shields %22$.1f:       %23$.f   (%24$.2f per PP)
+For [[encyclopedia ENC_SPECIES]]: %1%
+Total [[encyclopedia DAMAGE_TITLE]]: %2%    [[PC_DIRECT_WEAPON]] Shots: %3%
```
Comment (MatGB) [inline default/stringtables/en.txt:3977]: Direct shots doesn't seem to be working properly, possibly counting parts instead of ROF?

**Comment by dbenage-cx** [comment]
@MatGB Thanks for reviewing, will look into shots fired

For hull type, does not sound overly complicated to accomplish, will put on todo list ;)

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -4146,19 +4111,46 @@ Parts: %3%
...
+[[metertype METER_STRUCTURE]]: %4%    [[metertype METER_SHIELD]]: %5%
+[[metertype METER_SPEED]]: %6%    [[metertype METER_FUEL]] [[METER_CAPACITY]]: %7%
+[[metertype METER_DETECTION]]: %8%    [[metertype METER_STEALTH]]: %9%
```
[the remark this answers is not in the document]
Comment (dbenage-cx) [reply default/stringtables/en.txt]: Fixed (and speed/detection too :/ )

File: default/stringtables/en.txt (hunk 1 above)
Comment (dbenage-cx) [reply default/stringtables/en.txt:3977]: Fixed

File: default/stringtables/en.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (MatGB) [reply default/stringtables/en.txt]: Um, not sure it is, I've recompiled from this branch and both are swapped in my display.

[the remark this answers is not in the document]
Comment (dbenage-cx) [reply default/stringtables/en.txt]: Fetch/Rebase? Both github and local show `[[metertype METER_DETECTION]]: %6%    [[metertype METER_STEALTH]]: %7%`

[the remark this answers is not in the document]
Comment (MatGB) [reply default/stringtables/en.txt]: Did both before compiling earlier, will do so again in a bit...

Wait, sorry, I've messed up my git repo and created a branch with the same name as your remote branch and it's throwing errors at me, which means my half hour compiling earlier was a complete waste. oops

**Comment by dbenage-cx** [comment]
- [x]  TODO Remove `ENC_SHIP_DESIGN_DESCRIPTION_STR` from all stringtables.

**Comment by MatGB** [comment]
OK, all now appears to compile, run and display correctly, apologies for my error earlier.

**Comment by dbenage-cx** [comment]
Rebased with changes for non-default stringtable updates

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -4149,19 +4114,46 @@ Parts: %3%
...
-Colonization Capacity: %15%    [[metertype METER_TROOPS]]: %16%
-Estimated combat strength (ignoring shields):  %19$.f   (%20$.2f per PP)
-Against enemy attack %21$.1f and shields %22$.1f:       %23$.f   (%24$.2f per PP)
+For [[encyclopedia ENC_SPECIES]]: %1%
+Total [[encyclopedia DAMAGE_TITLE]]: %2%    [[PC_DIRECT_WEAPON]] Shots: %3%
+[[metertype METER_STRUCTURE]]: %4%    [[metertype METER_SHIELD]]: %5%
+[[metertype METER_DETECTION]]: %6%    [[metertype METER_STEALTH]]: %7%
+[[metertype METER_SPEED]]: %8%    [[metertype METER_FUEL]] [[METER_CAPACITY]]: %9%
+Colonization [[METER_CAPACITY]]: %10%    [[metertype METER_TROOPS]]: %11%
+[[OBJ_FIGHTER]] [[encyclopedia DAMAGE_TITLE]]: %13%    [[ENC_SDD_HANGAR]]: %12%    [[ENC_SDD_BAY]]: %14%
```
Comment (geoffthemedio) [inline default/stringtables/en.txt:3982]: This pulls in strings that reference "Current Fighters" which doesn't make sense for a ship design article, which is not about an actual ship that could contain any fighters currently.

File: default/stringtables/en.txt [hunk 4]
```diff
@@ -4149,19 +4114,46 @@ Parts: %3%
...
+Estimated combat strength (ignoring shields):  %15$.f   (%16$.2f per PP)
+Against enemy attack %17$.1f and shields %18$.1f:       %19$.f   (%20$.2f per PP)
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: The meaning of this "against enemy attack" stat isn't clear.

File: universe/Ship.cpp [hunk 5]
```diff
@@ -431,6 +431,37 @@ float Ship::InitialPartMeterValue(MeterType type, const std::string& part_name)
     return 0.0f;
 }
 
+float Ship::CurrentPartClassMeterValue(MeterType type, ShipPartClass part_class) const {
```
Comment (geoffthemedio) [inline universe/Ship.cpp]: purpose of function not clear from its name

**Review by geoffthemedio — CHANGES_REQUESTED** [review]

File: default/stringtables/en.txt (hunk 4 above)
Comment (dbenage-cx) [reply default/stringtables/en.txt]: Adding some formatting to help clarify.

File: universe/Ship.cpp (hunk 5 above)
Comment (dbenage-cx) [reply universe/Ship.cpp]: any suggestions for a name shorter than `SumCurrentPartMeterValuesForPartClass` highly welcome

File: default/stringtables/en.txt (hunk 3 above)
Comment (dbenage-cx) [reply default/stringtables/en.txt:3982]: Fixed

File: default/stringtables/en.txt [hunk 6]
```diff
@@ -7809,10 +7801,23 @@ Direct Weapon
 PC_FIGHTER_HANGAR
 Hangar
 
+PC_FIGHTER_HANGAR_DESC
+'''Hangars are the storage, maintenance, and resupply areas for fighters.
+Since each hangar is built in support of other hangars on the ship, a ship is limited to one specific hangar type.
+
+After combat, fighters need to return to a hangar for repairs, fuel, and rearmament.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: remove "need to"

File: default/stringtables/en.txt [hunk 7]
```diff
@@ -7809,10 +7801,23 @@ Direct Weapon
 PC_FIGHTER_HANGAR
 Hangar
 
+PC_FIGHTER_HANGAR_DESC
+'''Hangars are the storage, maintenance, and resupply areas for fighters.
+Since each hangar is built in support of other hangars on the ship, a ship is limited to one specific hangar type.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Don't need the explanation "Since each hangar ... on the ship,". Would rephrase as just "A ship cannot have more than one type of hangar."

File: default/stringtables/en.txt [hunk 8]
```diff
@@ -7809,10 +7801,23 @@ Direct Weapon
...
+
+After combat, fighters need to return to a hangar for repairs, fuel, and rearmament.
+Any fighter that is unable to dock at a hangar will likely be lost to the cold realities of space.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Rephrase without flavour text... "If a carrier is destroyed in combat, any of its fighters that survive the battle will be unable to dock after combat, and will also be lost."

File: default/stringtables/en.txt [hunk 9]
```diff
@@ -7809,10 +7801,23 @@ Direct Weapon
...
+'''
+
 # Ship part classes
 PC_FIGHTER_BAY
 Launch Bay
 
+PC_FIGHTER_BAY_DESC
+'''Launch bays are the access point for fighters to enter or exit a ship.
+Each launch bay allows a certain number of fighters to launch each combat round.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: don't need "certain"

**Review by geoffthemedio — CHANGES_REQUESTED** [review]

File: default/stringtables/en.txt (hunk 8 above)
Comment (dbenage-cx) [reply default/stringtables/en.txt]: Are fighters tied to a specific ship?  I was under impression they docked wherever there was available room.

File: default/stringtables/en.txt [hunk 10]
```diff
@@ -7803,10 +7803,10 @@ Hangar
 
 PC_FIGHTER_HANGAR_DESC
 '''Hangars are the storage, maintenance, and resupply areas for fighters.
-Since each hangar is built in support of other hangars on the ship, a ship is limited to one specific hangar type.
+A ship can not have more than one type of hangar.
 
-After combat, fighters need to return to a hangar for repairs, fuel, and rearmament.
-Any fighter that is unable to dock at a hangar will likely be lost to the cold realities of space.
+After combat, fighters return to a hangar for repairs, fuel, and rearmament.
+AIf a carrier is destroyed in combat, any of its fighters that survive the battle will be unable to dock after combat, and will also be lost.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: AIf?

File: default/stringtables/en.txt (hunk 8 above)
Comment (geoffthemedio) [reply default/stringtables/en.txt]: iirc they are specific to a particular ship

Comment (MatGB) [reply default/stringtables/en.txt]: Yes, we discussed doing otherwise but then you get sticky issues about crew species and other balance problems.

File: default/stringtables/en.txt (hunk 10 above)
Comment (dbenage-cx) [reply default/stringtables/en.txt]: Ugly way to destroy a carrier, fixed. (needs squash/fixup)

**Review by geoffthemedio — APPROVED** [review]

**Comment by geoffthemedio** [comment]
Maybe it's not related to this pull request, but the estimated combat strength values make no sense to me... Starting up a new game, and looking at the article for a random design, I see:
````
Colonization Capacity: 0    Troops: 0
Fighter Damage: 0    Hangar Capacity: 0    Launch Bay Capacity: 0
Estimated combat strength   (ignoring shields):  29   (0.25 per PP)
            (enemy with attack 3.1 and shields 0.0):  285   (2.38 per PP)
````
What does the (enemy with attack...) value mean? Is that just to compare this design to, or is it a rating for this ship against a particular example enemy?

**Comment by dbenage-cx** [comment]
Some rating against an enemy (clone of self except for particular stats).
Personally, I have mostly ignored these values when playing, so in favor of removing them if they are confusing.

**Comment by MatGB** [comment]
I can't remember when they were introduced, and I also mostly ignore them, but IIRC it was either @Morlic-fo or @Dilvish-fo that came up with the idea and some people seemed to really like it at the time. Anyone?

I'd say it needs either better explanation or removal but it's been there for ages now in some form.

**Comment by geoffthemedio** [comment]
if it has nothing to do with this pull request, then ok...

**Comment by dbenage-cx** [comment]
TY for reviews
